### ISSUE TITLE
sub-galleries should be generated with a separate "gallery rst" file

### ISSUE THREAD

**Issue by choldgraf** [sphinx-gallery/sphinx-gallery#413]
One of the challenges with sphinx-gallery is that it doesn't give a lot of flexibility with how you slice and dice the gallery. Since we don't allow for things like directives, you can't say "show the example gallery for these python files here, and those python files there" super easily.

One way to get around this would be to build all of the gallery code for sub-galleries (e.g. those within sub-folders of the main gallery folder) as separate text files, and then **include** them in the built gallery (minus the label). This would let users add the sub-galleries in other places if they so-wished.

## An example

For example, in the SG repo, we have the `sin_func` sub-folder in [the examples folder](https://github.com/sphinx-gallery/sphinx-gallery/tree/master/examples). When the gallery is built, the gallery for the `sin_func` folder is placed within `index.rst` in a block like this:

```
...other gallery code...

.. _sin_examples:

The sin function
----------------

Specific examples for the sin function. Saved in the examples
directory in a subfolder.



.. raw:: html

    <div class="sphx-glr-thumbcontainer" tooltip="A simple example of the plot of a sin function in order to test the autonomy of the gallery.">

.. only:: html

    .. figure:: /auto_examples/sin_func/images/thumb/sphx_glr_plot_sin_thumb.png

        :ref:`sphx_glr_auto_examples_sin_func_plot_sin.py`

.. raw:: html

    </div>


.. toctree::
   :hidden:

   /auto_examples/sin_func/plot_sin

.. raw:: html

    <div class="sphx-glr-thumbcontainer" tooltip="A simple example of the plot of a sin function on a black background">

...rest of gallery code...

```

This means that the rST for the `sin_func` gallery is interwoven with the gallery for the entire `examples` folder. What if the user wanted to highlight only the `sin_func` folder in a location in their docs? If instead we put the `sin_func` rST in a separate folder (e.g. in `sin_func/index.txt`) and then included it in the main `index.rst` file like so:

```
...other gallery code...

.. _sin_examples:

.. include:: sin_func/index.txt

...rest of gallery code...
```

then people could choose to include the `sin_func` mini-gallery wherever they like.

Does that make sense? I think it'd be a relatively simple patch since these are all auto-generated. What do folks think?

**Comment by Titan-C** [comment]
Yes and no. Subgalleries can have its own rst file, great, everything
becomes more modular. The no is just a minor annoyance regarding header
levels. Indeed you can import the rst from the sub-galleries, and
directly from anywhere as all paths are in absolute to the domain name
it would just work. But depending on where you import you might have
problems on the header levels. Sub-galleries are second to top level,
but if you want to import elsewhere you probably are going to have a
collision.

Honestly we already have this problem nowadays with the
mini-galleries. We underline headers with the caret '^' hoping that to
be the lowest level possible and not to generate to much conflict with
the user document hierarchy.

**Comment by choldgraf** [comment]
that's a good point! I guess the only solution would be to document best-practices. Are we still fairly set on not supporting sphinx directives? That'd be another potential solution to the problem, e.g.:

```
.. sphinx_gallery::
   :folder: ../examples/sin_func
```

or something like this

**Comment by lucyleeow** [comment]
I think this can be closed with #685
